src/api/app/services/notification_service/notifier.rb

```diff
@@ -0,0 +1,52 @@
+module NotificationService
+  class Notifier
+    EVENTS_TO_NOTIFY = ['Event::RequestStatechange',
+                        'Event::RequestCreate',
+                        'Event::ReviewWanted',
+                        'Event::CommentForProject',
+                        'Event::CommentForPackage',
+                        'Event::CommentForRequest'].freeze
+    CHANNELS = [:web, :rss].freeze
+
+    def initialize(event)
+      @event = event
+    end
+
+    def call
+      return unless @event.eventtype.in?(EVENTS_TO_NOTIFY)
+
+      CHANNELS.each do |channel|
+        @event.subscriptions(channel).each do |subscription|
+          create_notification_per_subscription(subscription, channel)
+        end
+      end
+    rescue StandardError => e
+      Airbrake.notify(e, event_id: @event.id)
+    end
+
+    private
+
+    def create_notification_per_subscription(subscription, channel)
+      return unless create_notification?(subscription.subscriber, channel)
+
+      klass = channel == :web ? 'Web' : 'RSS'
+      "NotificationService::#{klass}Channel".constantize.new(subscription, @event).call
+    end
+
+    def create_notification?(subscriber, channel)
+      return false if subscriber&.away?
+      return false if channel == :rss && !subscriber.try(:rss_token)
```

Comment:
you could follow the example from line 37 and use `subscriber&.rss_token`, right?

Reply:
Yes, I wanted to try a small change and see if Code Climate accepted it. I'll address the other one as well.